core/src/test/java/org/apache/iceberg/TestSortOrder.java

```diff
@@ -342,6 +343,13 @@ public void testVariantUnsupported() {
         .hasMessage("Unsupported type for identity: variant");
   }
 
+  @Test
+  public void testUnknownSupported() {
+    Schema v3Schema = new Schema(Types.NestedField.optional(1, "u", Types.UnknownType.get()));
+
+    SortOrder.builderFor(v3Schema).withOrderId(10).asc("u").build();
```

Comment:
Why is this allowed?

Reply:
I agree that this doesn't make sense, but as @HonahX pointed out here https://github.com/apache/iceberg/pull/12012#discussion_r1924520051, the `Unknown` type is a `PrimitiveType` and is accepted by the `IdentityTransform`. It doesn't make sense, but it is the same as sorting on a column that has many `null` values. I think we trust the user to make a wise decision and not to sort on an `UnknownType`, until it evolved in an actual type.

Reply:
:shrug: Since they only exist in variant doesn't that never come up?

Reply:
it looks a bit weird if there's no assertion check in a test method. So I would either add an assertion around the sort order or just add a `assertThatNoException().isThrownBy(...)`

Reply:
Good one, I've added it 👍